python/pyspark/ml/param/_shared_params_code_gen.py

```diff
@@ -47,7 +47,7 @@ def _gen_param_header(name, doc, defaultValueStr):
     """
     template = '''class Has$Name(Params):
     """
-    Mixin for param $name: $doc.
+    Mixin for param $name: $doc
     """
```

Comment:
Thanks for your PR. I think we should resolve the dot problem at [L107](https://github.com/holdenk/spark/blob/SPARK-9774-add-python-api-for-ml-regression-isotonicregression/python/pyspark/ml/param/_shared_params_code_gen.py#L107) rather than removing dot here, otherwise some classes in shared.py will lack the dot in the end of the line such as [```HasMaxIter```](https://github.com/apache/spark/pull/8214/files#diff-3d1fb305acc7bab18e5d91f2b69018c7R25).
Because of not all elements in [```shared```](https://github.com/holdenk/spark/blob/SPARK-9774-add-python-api-for-ml-regression-isotonicregression/python/pyspark/ml/param/_shared_params_code_gen.py#L107)  array follow the same rule of whether to append dot at the end of doc filed. We should remove the dot at the end of doc field for ```probabilityCol```, ```stepSize```, ```handleInvalid```, ```elasticNetParam```, ```standardization```, ```thresholds``` and ```weightCol``` in the ```shared``` array and keep the dot in [L50](https://github.com/holdenk/spark/blob/SPARK-9774-add-python-api-for-ml-regression-isotonicregression/python/pyspark/ml/param/_shared_params_code_gen.py#L50).
This is not caused by your PR, but it will be glad that you can resolve it.